### PR TITLE
Move "migrated" headers to context items.

### PR DESCRIPTION
**Pull request by terencefan**
According to David's suggestion.

Relate to #739

### CODE REVIEW

**Comment by vicancy** [comment]
I think in `Header` is better as we'd like user to get an explicit idea from `HttpContext` from inside the `Hub` to know if the connection is a migrated one? Put it into `Items` sounds obscure to users.

**Comment by davidfowl** [comment]
Lets change this to a feature. It's much cleaner and type safe. Proposal

```C#
public interface IConnectionMigrationFeature
{
    string MigratedFrom { get; }
}
```

We implement this on the `ClientConnectionContext` and the user can get the type to detect if this connection was migrated.

File: src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs [hunk 1]
```diff
@@ -92,6 +87,13 @@ public ClientConnectionContext(OpenConnectionMessage serviceMessage, Action<Http
             Application = pair.Transport;
 
             HttpContext = BuildHttpContext(serviceMessage);
+
+            if (serviceMessage.Headers.TryGetValue(Constants.AsrsMigrateIn, out var val))
+            {
+                IsMigrated = true;
+                HttpContext.Items.Add(Constants.AsrsMigrateIn, val);
```
Comment (halter73) [inline src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: @davidfowl Didn't you want this on ConnectionContext.Items not HttpContext.Items?

```suggestion
                Items.Add(Constants.AsrsMigrateIn, val);
```

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: Yea but now I want a feature instead.

Comment (vicancy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: Feature looks good to me, maybe we make it `internal` first.

Comment (halter73) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: Why would we make the feature internal? Are you talking about the feature interface itself?

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: Right, it needs to be public.

Comment (vicancy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: `internal` when it is not yet feature-ready, and switch to `public` when the feature is ready

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs]: Oh sure, we shouldn't ship an SDK with this feature being internal, that's all I'm saying.

**Pull request closed without merging**